app/screens/send-bitcoin-screen/payment-details/index.ts

```diff
@@ -0,0 +1,91 @@
+import { WalletCurrency } from "@app/graphql/generated"
+import { ConvertPaymentAmount, PaymentDetail } from "./index.types"
+import { ValidPaymentDestination } from "../send-bitcoin-reducer"
+import {
+  CreateAmountOnchainPaymentDetails,
+  CreateNoAmountOnchainPaymentDetails,
+} from "./onchain-payment-details"
+import { CreateIntraledgerPaymentDetails } from "./intraledger-payment-details"
+import {
+  CreateAmountLightningPaymentDetails,
+  CreateLnurlPaymentDetails,
+  CreateNoAmountLightningPaymentDetails,
+} from "./lightning-payment-details"
+
+export * from "./onchain-payment-details"
+export * from "./intraledger-payment-details"
+export * from "./index.types"
+
+export const CreatePaymentDetails = (
+  validPaymentDestination: ValidPaymentDestination,
+  convertPaymentAmount: ConvertPaymentAmount,
+): PaymentDetail<WalletCurrency> => {
+  switch (validPaymentDestination.paymentType) {
+    case "onchain":
+      if (validPaymentDestination.amount) {
+        return CreateAmountOnchainPaymentDetails({
+          address: validPaymentDestination.address,
+          destinationSpecifiedAmount: {
+            amount: validPaymentDestination.amount,
+            currency: "BTC",
+          },
+          convertPaymentAmount,
+          destinationSpecifiedMemo: validPaymentDestination.memo,
+          unitOfAccount: "USD",
```

Comment:
They create the same type (PaymentDetails which has a unitOfAccountAmount) but they are constructor functions that take in different arguments because an invoice with an amount is not capable of specifying the desired unitOfAccountAmount, the amount comes from the invoice amount and then it is converted to the desired unit of account. Whereas an invoice without an amount will determine its settlement amount from the desired unit of account amount entered.